graphql-dgs-codegen-core/src/main/kotlin/com/netflix/graphql/dgs/codegen/generators/java/ReservedKeywordSanitizer.kt

```diff
@@ -0,0 +1,35 @@
+/*
+ *
+ *  Copyright 2020 Netflix, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package com.netflix.graphql.dgs.codegen.generators.java
+
+class ReservedKeywordSanitizer {
+
+    companion object {
+        private val reservedKeywords = setOf("import", "_")
+        private const val prefix = "_"
+
+        fun sanitize(originalName: String): String {
```

Comment:
Have you looked at [NameAllocator](https://square.github.io/javapoet/javadoc/javapoet/com/squareup/javapoet/NameAllocator.html)? It will generate identifiers that avoid conflicts with keywords:

```kotlin
   val nameAllocator = NameAllocator()
   nameAllocator.newName("public") // public_
   nameAllocator.newName("package") // package_
```

Reply:
When running on JDK9 or above, "_" is also treated as a keyword, as NameAllocator delegates to [SourceVersion.isKeyword](https://github.com/openjdk/jdk/blob/master/src/java.compiler/share/classes/javax/lang/model/SourceVersion.java#L433)

Reply:
Great suggestion, I wasn't aware of NameAllocator.

Reply:
Since there was a blocking issue requiring this PR, and the NameAllocator requiring more involved changes I deferred using that for later. 
The `_` postfix isn't great because that also shows up in the generated method names, even for non-reserved names, which isn't as nice. Will look at this further later since it does do more than the implementation I added.

#20  for Kotlin specifically, since it's generator is only filtering `_` names now. Other keywords didn't seem to be an issue, but the `root` and `parent` methods are.